### DISCUSSION TITLE
ED Resources are not loading after local installation.

### DISCUSSION THREAD

**Discussion by Kovendan03** [Sunbird-Ed/Community#209] [Installation] [answered]
Hi Team,

I have followed the instructions for installing the ED locally using the documentation - https://github.com/Sunbird-Ed/SunbirdEd-portal. Installation was successful after modifications, but when i click the login getting the below error.

Error: Failed to lookup view "/root/SunbirdEd-portal/src/app/dist/index.ejs" in views directory "/root/SunbirdEd-portal/src/app/views"
    at Function.render (/root/SunbirdEd-portal/src/app/node_modules/express/lib/application.js:580:17)
    at ServerResponse.render (/root/SunbirdEd-portal/src/app/node_modules/express/lib/response.js:971:7)
    at renderDefaultIndexPage (/root/SunbirdEd-portal/src/app/routes/clientRoutes.js:257:11)
    at loadTenantFromLocal (/root/SunbirdEd-portal/src/app/routes/clientRoutes.js:286:5)
    at renderTenantPage (/root/SunbirdEd-portal/src/app/routes/clientRoutes.js:277:5)
    at Layer.handle [as handle_request] (/root/SunbirdEd-portal/src/app/node_modules/express/lib/router/layer.js:95:5)
    at next (/root/SunbirdEd-portal/src/app/node_modules/express/lib/router/route.js:137:13)
    at next (/root/SunbirdEd-portal/src/app/node_modules/express/lib/router/route.js:131:14)
    at next (/root/SunbirdEd-portal/src/app/node_modules/express/lib/router/route.js:131:14)
    at next (/root/SunbirdEd-portal/src/app/node_modules/express/lib/router/route.js:131:14)
    at next (/root/SunbirdEd-portal/src/app/node_modules/express/lib/router/route.js:131:14)
    at next (/root/SunbirdEd-portal/src/app/node_modules/express/lib/router/route.js:131:14)
    at next (/root/SunbirdEd-portal/src/app/node_modules/express/lib/router/route.js:131:14)
    at Route.dispatch (/root/SunbirdEd-portal/src/app/node_modules/express/lib/router/route.js:112:3)
    at Layer.handle [as handle_request] (/root/SunbirdEd-portal/src/app/node_modules/express/lib/router/layer.js:95:5)
    at /root/SunbirdEd-portal/src/app/node_modules/express/lib/router/index.js:281:22
    at param (/root/SunbirdEd-portal/src/app/node_modules/express/lib/router/index.js:354:14)
    at param (/root/SunbirdEd-portal/src/app/node_modules/express/lib/router/index.js:365:14)
    at Function.process_params (/root/SunbirdEd-portal/src/app/node_modules/express/lib/router/index.js:410:3)
    at next (/root/SunbirdEd-portal/src/app/node_modules/express/lib/router/index.js:275:10)
    at SendStream.error (/root/SunbirdEd-portal/src/app/node_modules/serve-static/index.js:121:7)
    at SendStream.emit (node:events:513:28)

Checked for the views directory but it is not available. Could you please help me on this as i have stuck on this and not able to proceed further. Also referred - https://github.com/orgs/Sunbird-Ed/discussions/115#discussioncomment-3677806

**Comment by gandham-santhosh** [comment]
@sharathkashyap @rajeevsathish Can you please help here

**Reply by rajeshkumaravel** [reply]
@Kovendan03 
Can you please check on environment variable `DEFAULT_CHANNEL: env.sunbird_default_channel,`
The default channel set should have tenant folder configured under `SunbirdEd-portal/src/app/tenant/DEFAULT_CHANNEL`.
By default `https://github.com/Sunbird-Ed/SunbirdEd-portal/tree/master/src/app/tenant/sunbird`; sunbird folder is configured for tenant.
Make a duplicate of sunbird folder under tenant and rename folder name accordingly.

CC: @sharathkashyap @rajeevsathish

**Comment by Kovendan03** [comment]
@rajeshkumaravel 

Thanks for your reply. I verified the environment variable DEFAULT_CHANNEL, it is set to 'sunbird', I have added to .bashrc file and sourced it 'export sunbird_default_tenant=sunbird'

Also, mentioned folder exists in the path.

/SunbirdEd-portal/src/app/tenant/sunbird# 

total 32
-rw-r--r-- 1 root root 3662 Sep 28 11:27 appLogo.png
drwxr-xr-x 2 root root 4096 Sep 28 11:27 css
drwxr-xr-x 3 root root 4096 Sep 28 11:27 fonts
drwxr-xr-x 3 root root 4096 Sep 28 11:27 images
-rw-r--r-- 1 root root 6993 Sep 28 11:27 index.html
drwxr-xr-x 2 root root 4096 Sep 28 11:27 js
drwxr-xr-x 2 root root 4096 Sep 28 11:27 vendor

I am able to see the below screen shot, when i click the links, getting the above error.

![image](https://user-images.githubusercontent.com/114154088/194477053-e2e9afca-b735-41d3-9542-a4bc87f8a6c5.png)

**Comment by rajeshkumaravel** [comment, marked as answer]
@Kovendan03 
- The links displayed in landing page (referring to above screenshot) are for examples; please edit `SunbirdEd-portal/src/app/tenant/sunbird/index.html` for appropriate action(s) / redirection.
- Tenant specific landing pages needs to be updated manually.
- You can try accessing `/explore` after the URL for explore page.

CC: @sharathkashyap @rajeevsathish @luckynvdu

**Comment by Kovendan03** [comment]
@rajeshkumaravel , I tried /explore after the URL, gut got error page.
![image](https://user-images.githubusercontent.com/114154088/194763690-4db54cad-84d7-48cb-b7ef-57fdd492d7d1.png)

Also, Tenant specific landing pages needs to be updated manually. --> please correct me if i am wrong, these specific pages redirection will be pertaining to our requirements?

**Reply by rajeshkumaravel** [reply]
@Kovendan03 
Looks like secret is not configured for express-session in server.js
Can you confirm  ```PORTAL_SESSION_SECRET_KEY: (env.sunbird_portal_session_secret && env.sunbird_portal_session_secret !== '')
  ? env.sunbird_portal_session_secret.split(',') : ''```
  this variable is set ?
  
  CC: @sharathkashyap @rajeevsathish @luckynvdu

**Comment by Kovendan03** [comment]
@rajeshkumaravel 

Value is not set --> PORTAL_SESSION_SECRET_KEY: (env.sunbird_portal_session_secret && env.sunbird_portal_session_secret !== ''),  where i can generate this session key

**Reply by rajeshkumaravel** [reply]
@Kovendan03 
The environment value can be set at sunbird-devops repo
```https://github.com/project-sunbird/sunbird-devops/blob/master/ansible/roles/stack-sunbird/templates/sunbird_player.env#L158```
```https://github.com/project-sunbird/sunbird-devops/blob/master/private_repo/ansible/inventory/dev/Core/secrets.yml#L140```
For local instance you can modify with custom value with comma separated.
Ex: ```PORTAL_SESSION_SECRET_KEY:sunbird,VALID_UUID```

CC: @sharathkashyap @rajeevsathish @luckynvdu

**Comment by Kovendan03** [comment]
@rajeshkumaravel 
I have generated a uuid and used in PORTAL_SESSION_SECRET_KEY: (env.sunbird_portal_session_secret && env.sunbird_portal_session_secret !== 'sunbird, b8120357-3f86-4d33-89c4-b2b95b45db8e'), but still getting the same failure.

**Reply by rajeevsathish** [reply]
@Kovendan03 can we have a one on one call and see what is the issue in details so that we can put the findings in the community so that it will help others if such issue is encountered you can connect with @rajeevsathish or @rajeshkumaravel for more details

**Reply by sagivpn** [reply]
@rajeevsathish Thanks for your response. Kindly let us know how you want to connect along with @Kovendan03 . He will share the invite accordingly.

**Reply by Kovendan03** [reply]
@rajeevsathish  @rajeshkumaravel , i have logged in dscord portal from morning today, please let me know if you available we can have a call.

**Comment by Kovendan03** [comment]
@gandham-santhosh @rajeevsathish @rajeshkumaravel , i have logged in dscord portal from morning today, please let me know if you available we can have a call.

**Comment by Kovendan03** [comment]
@rajeevsathish will schedule a google meet to discuss further.

**Reply by rajeevsathish** [reply]
Sure please ping me the link will join

**Comment by Kovendan03** [comment]
@rajeevsathish, I have scheduled a meeting invite and share the link. please join.

**Comment by Kovendan03** [comment]
@rajeevsathish , I have updated the Learner service URL in the configuration. Below page is displayed.

![image](https://user-images.githubusercontent.com/114154088/196352873-06b00808-174b-4253-84af-e1f5e43ac377.png)

**Reply by rajeevsathish** [reply]
this is not the config error but some of the form not updated and also the some of the environment variable is not set properly so this is the UI from SunbirdEd UI but none of the form data is loading

**Comment by Kovendan03** [comment]
I have set the environment variables for lern services, does the auth token has to be generated from the tenant or it has to be requested? Also, any resolution on the forms loading?

**Reply by sharathkashyap** [reply]
@Kovendan03 There are too many questions being asked and answered in this thread. I suggest a thread to be kept for one issue. Locking this thread
